src/server/tiering/op_manager.cc

```diff
@@ -102,11 +102,21 @@ void OpManager::ProcessStashed(EntryId id, unsigned version, DiskSegment segment
 }
 
 void OpManager::ProcessRead(size_t offset, std::string_view value) {
+  util::FiberAtomicGuard guard;  // atomically update items, no in-between states should be possible
   ReadOp* info = &pending_reads_.at(offset);
 
+  // Reorder base read (offset 0) to be last, so reads for defragmentation are handled last
```

Comment:
can you please explain this?

Reply:
If we already have a page read for defragmentation pending and some other read for the sub-segment is enqueued, we first must handle the sub-segment read, only then the full page read

Reply:
I suggest copying this explanation into the comment